mozilla/addons-server
# Make up go brrrrrrrr

Fixes: mozilla/addons#15066
Fixes: mozilla/addons#15046


### Description

This PR cuts out most of the remaining slowness from make up in development mode by:

- removing the locale/assets/source stages from development
- removing and simplifying docker volumes
- etc etc

### Context

With this PR make up in dev mode will be much more efficient, though slightly different than production mode. Running in production mode is now trivial so this is probably an okay tradeoff. If you want to verify  a feature or develop like you're in prod, you can do so. But we will save a lot of time by making up fast.

This PR also fixes race conditions we had during docker build by:
- ensuring only 1 of development or production dependency stages is installed per build.
- ensuring no volumes are shared by web/worker. By using anon volumes we now will have 1 for each. Not a big deal as they will be cleaned up automatically

### Testing

Run make up in development and production mode by specigying the docker target

```bash
make up DOCKER_TARGET=production
```

```bash
make up DOCKER_TARGET=development
```

You should be able to:
- load the app, no static files missing
- load i118n json files (in prod they will be real in dev they will be stubs)
- upload/download addon artifacts

### Seeding

This PR also fixes a bug where seeding the db does not synchronize the storage with the db state. Now it does. So running 

```bash
./manage.py seed_data
```

Should clear the DB, seed it and sync the data and storage directories.

### Checklist
